File: src/main/java/pl/training/workshop/m7/s13_godclass/step3/LegacyDb.java
package pl.training.workshop.m7.s13_godclass.step3;

import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;

/** "Baza danych" starego systemu - globalne, mutowalne mapy. Krok 3: rezerwacje jako typ Booking. */
public final class LegacyDb {
    public static final Map<String, Object[]> SCREENINGS = new LinkedHashMap<>();
    public static final Map<String, Booking> BOOKINGS = new LinkedHashMap<>();
    public static final Map<String, Integer> LOYALTY = new HashMap<>();
    public static int sequence = 1;

    private LegacyDb() {
    }

    public static void clear() {
        SCREENINGS.clear();
        BOOKINGS.clear();
        LOYALTY.clear();
        sequence = 1;
        LegacyMailer.SENT.clear();
        LegacyPaymentGateway.CHARGES.clear();
    }
}

File: typescript/src/workshop/legacy/LegacyDb.ts
import { LegacyMailer } from './LegacyMailer.js';
import { LegacyPaymentGateway } from './LegacyPaymentGateway.js';

// "Baza danych" starego systemu - globalne, mutowalne mapy.
export class LegacyDb {
  static readonly SCREENINGS = new Map<string, unknown[]>();
  static readonly BOOKINGS = new Map<string, unknown[]>();
  static readonly LOYALTY = new Map<string, number>();
  static sequence = 1;

  private constructor() {}

  static clear(): void {
    LegacyDb.SCREENINGS.clear();
    LegacyDb.BOOKINGS.clear();
    LegacyDb.LOYALTY.clear();
    LegacyDb.sequence = 1;
    LegacyMailer.SENT.length = 0;
    LegacyPaymentGateway.CHARGES.length = 0;
  }
}
